packages/perseus/src/widgets/interactive-graphs/interactive-graph-question-builder.ts

```diff
@@ -318,6 +323,17 @@ class InteractiveGraphQuestionBuilder {
             showPoint2: options?.showPoint2 ?? false,
             color: options?.color ?? "grayH",
             lineStyle: options?.lineStyle ?? "solid",
+            labels: options?.labels
+                ? options.labels.map((label) => ({
+                      type: "label",
+                      coord:
+                          label.coord ??
+                          ([...kline.midpoint([point1, point2])] as Coord),
+                      text: label.text,
+                      color: options.color ?? "grayH",
+                      size: label.size ?? "medium",
+                  }))
+                : [],
```

Comment:
I think you could avoid the ternary operator and the duplicate reference to `options.labels` by doing this:

```ts
labels: (options?.labels ?? []).map((label) => ({
  // ...
})),
```

or by adding a default argument for `options` - then you wouldn't need all the `?.`s.

```ts
addLockedLine(
  point1: vec.Vector2,
  point2: vec.Vector2,
  options: {
    // ...
  } = {},
): InteractiveGraphQuestionBuilder {
  const {labels = []} = options;
  const line: LockedLineType = {
    // ...
    labels: labels.map((label) => ({
      // ...
    })),
  };
  // ...
}
```